ehcache-impl/src/main/java/org/ehcache/config/builders/UserManagedCacheBuilder.java

```diff
@@ -166,175 +167,178 @@ private UserManagedCacheBuilder(UserManagedCacheBuilder<K, V, T> toCopy) {
-      this.eventDispatcher = new CacheEventDispatcherImpl<>(unOrderedExecutor, orderedExecutor);
-    }
-    eventDispatcher.setStoreEventSource(store.getStoreEventSource());
+      if (this.eventDispatcher instanceof DisabledCacheEventNotificationService && (orderedExecutor != null & unOrderedExecutor != null)) {
+        this.eventDispatcher = new CacheEventDispatcherImpl<>(unOrderedExecutor, orderedExecutor);
+      }
+      eventDispatcher.setStoreEventSource(store.getStoreEventSource());
 
-    ResilienceStrategy<K, V> resilienceStrategy;
-    if (cacheLoaderWriter == null) {
-      resilienceStrategy = new RobustResilienceStrategy<>(new DefaultRecoveryStore<>(store));
-    } else {
-      resilienceStrategy = new RobustLoaderWriterResilienceStrategy<>(new DefaultRecoveryStore<>(store), cacheLoaderWriter);
-    }
+      ResilienceStrategy<K, V> resilienceStrategy;
+      if (cacheLoaderWriter == null) {
+        resilienceStrategy = new RobustResilienceStrategy<>(new DefaultRecoveryStore<>(store));
+      } else {
+        resilienceStrategy = new RobustLoaderWriterResilienceStrategy<>(new DefaultRecoveryStore<>(store), cacheLoaderWriter);
+      }
 
-    if (persistent) {
-      DiskResourceService diskResourceService = serviceLocator
+      if (persistent) {
+        DiskResourceService diskResourceService = serviceLocator
           .getService(DiskResourceService.class);
-      if (diskResourceService == null) {
-        throw new IllegalStateException("No LocalPersistenceService could be found - did you configure one?");
-      }
+        if (diskResourceService == null) {
+          throw new IllegalStateException("No LocalPersistenceService could be found - did you configure one?");
+        }
 
-      PersistentUserManagedEhcache<K, V> cache = new PersistentUserManagedEhcache<>(cacheConfig, store, resilienceStrategy, diskResourceService, cacheLoaderWriter, eventDispatcher, id);
-      registerListeners(cache, serviceLocator, lifeCycledList);
-      for (LifeCycled lifeCycled : lifeCycledList) {
-        cache.addHook(lifeCycled);
-      }
-      return cast(cache);
-    } else {
-      InternalCache<K, V> cache = new Ehcache<>(cacheConfig, store, resilienceStrategy, eventDispatcher, getLoggerFor(Ehcache.class));
-      registerListeners(cache, serviceLocator, lifeCycledList);
-      for (LifeCycled lifeCycled : lifeCycledList) {
-        (cache).addHook(lifeCycled);
+        PersistentUserManagedEhcache<K, V> cache = new PersistentUserManagedEhcache<>(cacheConfig, store, resilienceStrategy, diskResourceService, cacheLoaderWriter, eventDispatcher, id);
+        registerListeners(cache, serviceLocator, lifeCycledList);
+        for (LifeCycled lifeCycled : lifeCycledList) {
+          cache.addHook(lifeCycled);
+        }
+        return cast(cache);
+      } else {
+        InternalCache<K, V> cache = new Ehcache<>(cacheConfig, store, resilienceStrategy, eventDispatcher, getLoggerFor(Ehcache.class));
```

Comment:
The `getLoggerFor(Ehcache.class)` call here is suspicious. That method looks like it should be replaced with use of this new logging prefix scheme.

Reply:
replaced `getLoggerFor(Ehcache.class)` with `EhcachePrefixLoggerFactory.getLogger(Ehcache.class)`

Reply:
What about the method. Is it used any more?

Reply:
no usage to remove this method